### DISCUSSION TITLE
Directed Steiner Tree Algorithm

### DISCUSSION THREAD

**Discussion by sarahwooders** [networkx/networkx#6512] [Algorithms]
We have an application where edges are directed, and the costs are dependent on the direction. Would it be possible to add a directed Steiner Tree algorithm?

**Comment by boothby** [comment]
I'd love to see that!  The algorithm of [Watel and Wasser](https://cedric.cnam.fr/fichiers/art_3140.pdf) seems to be relatively uncomplicated with reasonable performance and approximation ratios.
